spec/src/main/asciidoc/programmatic_lookup.asciidoc

```diff
@@ -0,0 +1,36 @@
+//
+// Copyright (c) 2017 Contributors to the Eclipse Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+[[programmatic_lookup]]
+== MicroProfile Rest Client Programmatic Lookup
+
+Type Safe Rest Clients support both programmatic look up and CDI injection approaches for usage.  An implementation of MicroProfile Rest Client is expected to support both use cases.
+
+=== Sample Builder Usage
+
+[source, java]
+----
+public class SomeService {
+   public Response doWorkAgainstApi(URL apiUrl, ApiModel apiModel) {
+       RemoteApi remoteApi = TypeSafeClientBuilder.newBuilder()
```

Comment:
I think from looking at this example that the TSCB should actually be a TypeSafeClient.
Or proxy() should be renamed build() or similar.

Reply:
I agree that build() is better than proxy() - the user doesn’t really need to know or care that the underlying implementation is a dynamic proxy.

Reply:
Does `build()` still take the interface, or how does the interface go in?  The idea I had behind it is that you could chain a reusable client, e.g.:

```java
TypeSafeClientBuilder builder = TypeSafeClientBuilder.newBuilder().baseUrl(..);
builder.build(Interface1.class);
builder.build(Interface2.class);
```

Does that still make sense?

Reply:
+1 to build() instead of proxy()